### PR TITLE
fix(user-profiles): various regressions surrounding react-intl

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
A few bugs were reported which seem to all stem from changes we've made over the past few months with relation to react-intl. **I'm very unsure that my fixes are the ideal way to resolve these issues, so would appreciate a careful review from the original authors of the affected files!**

Specifically:
- adding a monitored trip only works for Saturday and Sunday
- monitored trip icons are vertical instead of horizontal
- adding a custom place doesn't work (can't select anything that `location-field` returns)

### CODE REVIEW

File: lib/components/user/monitored-trip/trip-basics-pane.tsx [hunk 1]
```diff
@@ -200,12 +200,11 @@ class TripBasicsPane extends Component<TripBasicsProps> {
                   : monitoredTrip[day]
                   ? 'bg-primary'
                   : ''
-                const notAvailableText = isDayDisabled
-                  ? intl.formatMessage(
-                      { id: 'components.TripBasicsPane.tripNotAvailableOnDay' },
-                      { repeatedDay: getFormattedDayOfWeekPlural(day, intl) }
-                    )
-                  : null
+                const notAvailableText = intl.formatMessage(
```
Comment (binh-dam-ibigroup) [inline lib/components/user/monitored-trip/trip-basics-pane.tsx:203]: What was the bug that you observed here? It looks like we are currently showing a popup that says "Trip not available on <day>" when the user hovers on a day for which a trip is available, but this change does not seem to fix this particular issue.

File: lib/components/user/places/place-editor.tsx [hunk 2]
```diff
@@ -64,15 +64,18 @@ class PlaceEditor extends Component<
...
 > {
-  _handleLocationChange = ({
-    location
-  }: {
-    location: {
-      lat: number
-      lon: number
-      name: string
+  _handleLocationChange = (
+    _: unknown, // Ignore intl object. TODO: localize name?
```
Comment (binh-dam-ibigroup) [inline lib/components/user/places/place-editor.tsx:68]: Use `IntlShape` for the arg type.

File: lib/components/user/places/place-editor.tsx [hunk 3]
```diff
@@ -64,15 +64,18 @@ class PlaceEditor extends Component<
...
+    {
+      location
+    }: {
+      location: {
```
Comment (binh-dam-ibigroup) [inline lib/components/user/places/place-editor.tsx:72]: For `location`, you can use the `Location` type from `@opentripplanner/types`.

**Review by binh-dam-ibigroup — COMMENTED** [review]
See the different comments

File: lib/components/user/monitored-trip/trip-basics-pane.tsx (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/components/user/monitored-trip/trip-basics-pane.tsx:203]: The bug here was that only Saturday and Sunday was showing up. I think that might be related to the bug you describe, but I think it's more serious to not have the days show then to not have them show in the tooltip

File: lib/components/user/places/place-editor.tsx (hunk 2 above)
Comment (miles-grant-ibigroup) [reply lib/components/user/places/place-editor.tsx]: Good call thanks. Refactored in 71c5a88

File: lib/components/user/places/place-editor.tsx (hunk 3 above)
Comment (miles-grant-ibigroup) [reply lib/components/user/places/place-editor.tsx]: Added in 71c5a88!

File: lib/components/user/monitored-trip/trip-basics-pane.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/components/user/monitored-trip/trip-basics-pane.tsx:203]: Okay, I see. Can you try the original code on this block and remove the `notAvailableText &&` condition in the `return` statement below? I think that's where things broke originally. In `dev` it reads that if a trip **is** available on a particular day, `notAvailable` would be `null` and the label/checkbox for the day is thus not shown.

File: lib/components/user/places/place-editor.tsx [hunk 4]
```diff
@@ -64,15 +65,10 @@ class PlaceEditor extends Component<
...
-    location
-  }: {
-    location: {
-      lat: number
-      lon: number
-      name: string
-    }
-  }) => {
+  _handleLocationChange = (
+    _: IntlShape, // Ignore intl object. TODO: localize name?
```
Comment (binh-dam-ibigroup) [inline lib/components/user/places/place-editor.tsx:69]: What do you mean by "localize name"?

File: lib/components/user/monitored-trip/trip-basics-pane.tsx (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/components/user/monitored-trip/trip-basics-pane.tsx:203]: Screen Shot 2022-04-19 at 10 38 49 AM

Switching to what you describe causes this to happen for all trips. Not sure if there's more refactoring you want to do here or if you'd rather move this debugging to a different PR

File: lib/components/user/places/place-editor.tsx (hunk 4 above)
Comment (miles-grant-ibigroup) [reply lib/components/user/places/place-editor.tsx]: I've clarified the comment in 36909e9b441bf6b9dc7bea471474ca1d80b39ee2. I'm not sure if this makes sense to do, but thought I could clarify what the `intl` object could possibly be used for one day

File: lib/components/user/monitored-trip/trip-basics-pane.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/components/user/monitored-trip/trip-basics-pane.tsx:203]: I'm not seeing the blanket unavailability.

Comment (binh-dam-ibigroup) [reply lib/components/user/monitored-trip/trip-basics-pane.tsx:203]: Again, remove the condition on original line 210, and revert original lines 203-208.

Comment (miles-grant-ibigroup) [reply lib/components/user/monitored-trip/trip-basics-pane.tsx]: Resolved in ec4dd4ab

**Comment by miles-grant-ibigroup** [comment]
Things seem to be working now

**Pull request merged**